api/scripts/prod/generate-knowledge-element-snapshots-for-active-campaigns.js

```diff
@@ -0,0 +1,113 @@
+    --help ou -h                : Affiche l'aide`);
+}
+
+function _validateAndNormalizeArgs(commandLineArgs) {
+  if (commandLineArgs.find((commandLineArg) => (commandLineArg === '--help' || commandLineArg === '-h'))) {
+    _printUsage();
+    process.exit(0);
+  }
+  const commandLineArgsLength = commandLineArgs.length;
+  const maxSnapshotCountIndicatorIndex = commandLineArgs.findIndex((commandLineArg) => commandLineArg === '--maxSnapshotCount');
+  let maxSnapshotCount;
+  if (maxSnapshotCountIndicatorIndex === -1 || maxSnapshotCountIndicatorIndex + 1 >= commandLineArgsLength) {
+    maxSnapshotCount = DEFAULT_MAX_SNAPSHOT_COUNT;
+  } else {
+    maxSnapshotCount = parseInt(commandLineArgs[maxSnapshotCountIndicatorIndex + 1]);
+    if (isNaN(maxSnapshotCount)) {
+      maxSnapshotCount = DEFAULT_MAX_SNAPSHOT_COUNT;
+    }
+    if (maxSnapshotCount <= 0) {
+      throw new Error(`Nombre max de snapshots ${commandLineArgs[maxSnapshotCountIndicatorIndex + 1]} ne peut pas être inférieur à 1.`);
+    }
+  }
+  const concurrencyIndicatorIndex = commandLineArgs.findIndex((commandLineArg) => commandLineArg === '--concurrency');
+  let concurrency;
+  if (concurrencyIndicatorIndex === -1 || concurrencyIndicatorIndex + 1 >= commandLineArgsLength) {
+    concurrency = DEFAULT_CONCURRENCY;
+  } else {
+    concurrency = parseInt(commandLineArgs[concurrencyIndicatorIndex + 1]);
+    if (isNaN(concurrency)) {
+      concurrency = DEFAULT_CONCURRENCY;
+    }
+    if (concurrency <= 0 || concurrency > 10) {
+      throw new Error(`Concurrent ${commandLineArgs[concurrencyIndicatorIndex + 1]} ne peut pas être inférieur à 1 ni supérieur à 10.`);
+    }
+  }
+
+  return {
+    maxSnapshotCount,
+    concurrency,
+  };
+}
+
+async function getEligibleCampaignParticipations(maxSnapshotCount) {
+  return knex('campaign-participations').select('campaign-participations.userId', 'campaign-participations.sharedAt')
+    .leftJoin('knowledge-element-snapshots', 'knowledge-element-snapshots.userId', 'campaign-participations.userId')
+    .join('campaigns', 'campaigns.id', 'campaign-participations.campaignId')
+    .whereNull('campaigns.archivedAt')
+    .whereNotNull('campaign-participations.sharedAt')
+    .where((qb) => {
+      qb.whereNull('knowledge-element-snapshots.snappedAt')
+        .orWhereRaw('?? != ??', ['campaign-participations.sharedAt', 'knowledge-element-snapshots.snappedAt']);
+    })
+    .orderBy('campaign-participations.userId')
+    .limit(maxSnapshotCount);
+}
+
+async function generateKnowledgeElementSnapshots(campaignParticipationData, concurrency) {
+  return bluebird.map(campaignParticipationData, async (campaignParticipation) => {
+    const { userId, sharedAt: snappedAt } = campaignParticipation;
+    const knowledgeElements = await knowledgeElementRepository.findUniqByUserId({ userId, limitDate: snappedAt });
+    return knowledgeElementSnapshotRepository.save({ userId, snappedAt, knowledgeElements });
+  }, { concurrency });
```

Comment:
Juste pour une meilleure compréhension du code, il serait mieux de conserver le nom de variable `sharedAt` : 
```suggestion
  return bluebird.map(campaignParticipationData, async (campaignParticipation) => {
    const { userId, sharedAt } = campaignParticipation;
    const knowledgeElements = await knowledgeElementRepository.findUniqByUserId({ userId, limitDate: sharedAt });
    return knowledgeElementSnapshotRepository.save({ userId, snappedAt: sharedAt, knowledgeElements });
  }, { concurrency });
```

Reply:
fait